_posts/2024-08-14-adding-to-bioconda-quickguide.md

```diff
@@ -0,0 +1,468 @@
+
+However, as always, check other tools/packages for examples.
+
+Examples of small `build.sh` scripts from the four examples above:
+
+- [kallisto](https://github.com/bioconda/bioconda-recipes/blob/23fe8cc0729ff70883819a8d2b2fdfc4d1da1443/recipes/kallisto/build.sh) (make install).
+- [MALT](https://github.com/bioconda/bioconda-recipes/blob/23fe8cc0729ff70883819a8d2b2fdfc4d1da1443/recipes/malt/build.sh) (java jar file).
+- [metabinner](https://github.com/bioconda/bioconda-recipes/blob/23fe8cc0729ff70883819a8d2b2fdfc4d1da1443/recipes/metabinner/build.sh) (auxiliary scripts).
+- [phynder](https://github.com/bioconda/bioconda-recipes/blob/23fe8cc0729ff70883819a8d2b2fdfc4d1da1443/recipes/phynder/build.sh) (patching).
+- [grid](https://github.com/bioconda/bioconda-recipes/blob/23fe8cc0729ff70883819a8d2b2fdfc4d1da1443/recipes/grid/build.sh) (database files).
+
+To provide further guidance based on my experience:
+
+The `$PREFIX` variable corresponds to the the root of the conda environment that eventually gets made on a users system when they install the conda package.
+You can explore our own conda environments to see what the `$PREFIX` looks like by running `conda env list` to see all of our own conda environments, and changing into the one of the directory listed in there.
+They often will look very similar to Unix root directories, with folders such as `etc/`, `bin/`, `lib/`, `share/`, etc.
+for example, if we have an executable or scripts that need to go into `bin/`, we must copy this into `$PREFIX/bin`.
+For some tools we may have to copy other files into other directories, such as databases [^6], but this is less common.
+
+Another tricky thing is compiling of C++ code, which can be a bit of a pain.
+For reasons [^7], we need to use specific variables that point to the non-standard (it seems) places that conda stores its libraries and headers.
+These are described [here](https://bioconda.github.io/contributor/guidelines.html#c-c), and in particular for [zlib](https://bioconda.github.io/contributor/troubleshooting.html#zlib-errors).
+You often will need to patch the `make` files and other compilation related scripts to use these variables, and also to use the `--prefix=$PREFIX` flag when running `make install`.
+
+For all of the above, regardless of language, I recommend looking at the the [contributor guidelines](https://bioconda.github.io/contributor/guidelines.html).
+
+### Build testing
+
+Once we think we've got our `meta.yaml` and `build.sh` (if needed) files ready, we can now try to see if this works.
+
+We have two options here, either:
+
+- Open the pull request onto the main `bioconda-recipes` repository and see if it passes the tests there (slow).
+- Test it locally (less slow, but may not perfectly replicate the build).
```

Comment:
It seems it might make sense to reverse the order of these options, since you might often want to start with local testing as it is faster, and then complement with the slower CI test once local testing passes?

Reply:
Good point :+1:  flipped